src/block-management/block-holder.js

```diff
@@ -45,36 +62,109 @@ export class BlockHolder extends React.Component<PropsType, StateType> {
 		};
 	}
 
-	renderToolbarIfBlockFocused() {
-		if ( this.props.isSelected ) {
-			return (
-				<InlineToolbar
-					clientId={ this.props.clientId }
-					onButtonPressed={ this.props.onInlineToolbarButtonPressed }
-					canMoveUp={ this.props.canMoveUp }
-					canMoveDown={ this.props.canMoveDown }
-				/>
-			);
+	onFocus = ( event: NativeSyntheticEvent<NativeTouchEvent> ) => {
+		if ( event ) {
+			// == Hack for the Alpha ==
+			// When moving the focus from a TextInput field to another kind of field the call that hides the keyboard is not invoked
+			// properly, resulting in keyboard up when it should not be there.
+			// The code below dismisses the keyboard (calling blur on the last TextInput field) when the field that now gets the focus is a non-textual field
+			const currentlyFocusedTextInput = TextInputState.currentlyFocusedField();
+			if ( event.nativeEvent.target !== currentlyFocusedTextInput && ! TextInputState.isTextInput( event.nativeEvent.target ) ) {
+				TextInputState.blurTextInput( currentlyFocusedTextInput );
+			}
 		}
+		this.props.onSelect( this.props.clientId );
+	};
 
-		// Return empty view, toolbar won't be rendered
-		return <View />;
+	onInlineToolbarButtonPressed = ( button: number ) => {
+		switch ( button ) {
+			case InlineToolbarActions.UP:
+				this.props.moveBlockUp();
+				break;
+			case InlineToolbarActions.DOWN:
+				this.props.moveBlockDown();
+				break;
+			case InlineToolbarActions.DELETE:
+				this.props.removeBlock();
+				break;
+		}
+	};
+
+	insertBlocksAfter = ( blocks: Array<Object> ) => {
+		const order = this.props.getBlockIndex( this.props.clientId, this.props.rootClientId );
```

Comment:
Scratch that, I see this falls under the "Switch from passing order to getBlockIndex as props to avoid rerender BlockHolder on order change" that you mention in the PR description.

Regarding the `getBlockIndex()` method though, can we do the same treatment to it and remove the parameters since those are constants for each BlockHolder?

Reply:
We could but it's not the best practice since `getBlockIndex()` is a selector, and thus injected in our props with `withSelect`. If we do that:
```
withSelect( ( select, { clientId, rootClientId } ) => {
	const { getBlockIndex } = select( 'core/editor' );
	return {
		...
		getBlockIndex: () => getBlockIndex( clientId, rootClientId ),
	};
} )
```
Then our `getBlockIndex` props will be re-created on each props change, which isn't optimal if let's say, `getBlockIndex` is passed as a props to another child component as it will force a refresh.

Reply:
Some precision about my last comment which can be confusing:
1 - Passing down props to a child is not a good practice in general, except in very few cases such as HOCs, or some very generic components.
2 - The main reason we don't want to wrap `getBlockIndex` inside a simpler version without parameters is that defining any function inside `withSelect` is bad practice as a new function will be created each time and given as new props. And `BlockHolder` could skip rerendering if no props have actually changed.